pkgs/os-specific/linux/virtualbox/default.nix

```diff
@@ -14,6 +14,12 @@ stdenv.mkDerivation {
   makeFlags = [ "INSTALL_MOD_PATH=$(out)" ];
   installTargets = [ "install" ];
 
+  # VirtualBox 6.1.18 doesn't build with linux kernel 5.11.  This fix is based on
+  # the discussion at https://forums.virtualbox.org/viewtopic.php?f=7&t=101860
+  patches = lib.optionals (lib.versionAtLeast kernel.version "5.11" || lib.versionAtLeast "6.1.18" virtualbox.version) [
```

Comment:
Does this patch break when applied to other kernels? If not we should apply it all the time to prevent bitrot.

Reply:
@SuperSandro2000 I think it ought to work with other kernels, but I'm not sure.

My expectation is that this will almost certainly be fixed in a near-future release of VirtualBox, so we could remove it then.

Reply:
I think because you used atLeast this should be safe to leave it like it is. If you would have used exact matches we would surely forget to remove it.

Reply:
OK, great, thanks!  If I notice it break, I'll certainly come back and fix it.

Is this ready for merge, then?

Reply:
You are sure the condition is correct? Right now the patch is applied if either is true, for example also when kernel 5.10 is used with 6.1.18.

Reply:
Good call!  Ought to be `&&`; just pushed a fix.